eng/pipelines/templates/stages/archetype-java-release-batch.yml

```diff
@@ -0,0 +1,497 @@
+                    artifact: ${{parameters.ArtifactName}}
+                  - template: /eng/common/pipelines/templates/steps/sparse-checkout.yml
+                    parameters:
+                      SkipDefaultCheckout: true
+                      Paths:
+                        - sdk
+                        - common/perf-test-core
+                        - .github/CODEOWNERS
+                        - '!sdk/**/test-recordings'
+                        - '!sdk/**/session-records'
+                  - ${{ each artifact in parameters.Artifacts }}:
+                    - ${{if ne(artifact.skipPublishDocMs, 'true')}}:
+                      - download: current
+                        displayName: 'Download Artifact: ${{parameters.ArtifactName}}'
+                        artifact: ${{parameters.ArtifactName}}
+                      # Pull and build the docker image.
+                      - template: /eng/common/pipelines/templates/steps/update-docsms-metadata.yml
+                        parameters:
+                          PackageInfoLocations:
+                            - $(Pipeline.Workspace)/${{parameters.ArtifactName}}/PackageInfo/${{artifact.name}}.json
+                          WorkingDirectory: $(System.DefaultWorkingDirectory)
+                          TargetDocRepoOwner: 'Azure'
+                          TargetDocRepoName: ${{parameters.TargetDocRepoName}}
+                          Language: 'java'
+                          SparseCheckoutPaths:
+                            - docs-ref-services/
+                            - metadata/
+                          DocValidationImageId: "$(DocValidationImageId)"
+                      - template: /eng/pipelines/templates/steps/fetch-package-list.yml
+                        parameters:
+                          JavaDocJarLocation: "$(Pipeline.Workspace)/${{parameters.ArtifactName}}/${{artifact.groupId}}/${{artifact.name}}"
+                          ArtifactName: ${{artifact.name}}
+                          TargetBranch: $(TargetBranchName)
+                          DocRepoLocation: $(DocRepoLocation)
+                          TargetDocRepoName: ${{parameters.TargetDocRepoName}}
+                          TargetDocRepoOwner: "Azure"
+
+        - deployment: PublishDocs
+          displayName: Publish Docs to GitHubIO Blob Storage
+          condition: and(succeeded(), ne(variables['Skip.PublishDocs'], 'true'))
+          environment: githubio
+          dependsOn: PublishPackage
+          variables:
+            - template: /eng/pipelines/templates/variables/globals.yml
+          pool:
+            name: azsdk-pool-mms-win-2019-general
+            vmImage: MMS2019
+          strategy:
+            runOnce:
+              deploy:
+                steps:
+                  - template: /eng/common/pipelines/templates/steps/sparse-checkout.yml
+                  - ${{ each artifact in parameters.Artifacts }}:
```

Comment:
Same question here. I assume this will become an inner loop inside the scripts.

Reply:
Even if we don't do the loop fully inside we should still move from the template loop to an inline script loop to eliminate all the noise for the repeated steps in the DevOPs job.

Reply:
For this particular case the publish-blobs.yml is pretty generic and we can just download all the artifacts once and then call it once to upload everything.